diff --git a/media/ffvpx/libavcodec/mediacodecdec.c b/media/ffvpx/libavcodec/mediacodecdec.c
--- a/media/ffvpx/libavcodec/mediacodecdec.c
+++ b/media/ffvpx/libavcodec/mediacodecdec.c
@@ -36,9 +36,13 @@
 #include "avcodec.h"
 #include "codec_internal.h"
 #include "decode.h"
+#if CONFIG_H264_MEDIACODEC_DECODER_EXTRADATA
 #include "h264_parse.h"
 #include "h264_ps.h"
+#endif
+#if CONFIG_HEVC_MEDIACODEC_DECODER_EXTRADATA
 #include "hevc/parse.h"
+#endif
 #include "hwconfig.h"
 #include "internal.h"
 #include "fffjni.h"
@@ -127,7 +131,7 @@ done:
 }
 #endif
 
-#if CONFIG_H264_MEDIACODEC_DECODER
+#if CONFIG_H264_MEDIACODEC_DECODER_EXTRADATA
 static int h264_set_extradata(AVCodecContext *avctx, FFAMediaFormat *format)
 {
     int i;
@@ -193,7 +197,7 @@ done:
 }
 #endif
 
-#if CONFIG_HEVC_MEDIACODEC_DECODER
+#if CONFIG_HEVC_MEDIACODEC_DECODER_EXTRADATA
 static int hevc_set_extradata(AVCodecContext *avctx, FFAMediaFormat *format)
 {
     int i;
@@ -344,7 +348,11 @@ static av_cold int mediacodec_decode_init(AVCodecContext *avctx)
     case AV_CODEC_ID_H264:
         codec_mime = "video/avc";
 
+#if CONFIG_H264_MEDIACODEC_DECODER_EXTRADATA
         ret = h264_set_extradata(avctx, format);
+#else
+        ret = 0;
+#endif
         if (ret < 0)
             goto done;
         break;
@@ -353,7 +361,11 @@ static av_cold int mediacodec_decode_init(AVCodecContext *avctx)
     case AV_CODEC_ID_HEVC:
         codec_mime = "video/hevc";
 
+#if CONFIG_HEVC_MEDIACODEC_DECODER_EXTRADATA
         ret = hevc_set_extradata(avctx, format);
+#else
+        ret = 0;
+#endif
         if (ret < 0)
             goto done;
         break;
@@ -637,11 +649,11 @@ const FFCodec ff_ ## short_name ## _mediacodec_decoder = {
 };                                                                                             \
 
 #if CONFIG_H264_MEDIACODEC_DECODER
-DECLARE_MEDIACODEC_VDEC(h264, "H.264", AV_CODEC_ID_H264, "h264_mp4toannexb")
+DECLARE_MEDIACODEC_VDEC(h264, "H.264", AV_CODEC_ID_H264, NULL)
 #endif
 
 #if CONFIG_HEVC_MEDIACODEC_DECODER
-DECLARE_MEDIACODEC_VDEC(hevc, "H.265", AV_CODEC_ID_HEVC, "hevc_mp4toannexb")
+DECLARE_MEDIACODEC_VDEC(hevc, "H.265", AV_CODEC_ID_HEVC, NULL)
 #endif
 
 #if CONFIG_MPEG2_MEDIACODEC_DECODER
